google/cloud/pubsub/internal/batch_sink.h

```diff
@@ -38,6 +39,9 @@ class BatchSink {
  public:
   virtual ~BatchSink() = default;
 
+  /// Adds the message to the batch the message.
+  virtual void AddMessage(pubsub::Message m) = 0;
```

Comment:
This is the new `SaveMessage()`, right?

https://github.com/googleapis/google-cloud-cpp/blob/945644596dfc9a7a6bcc3dac484f9c2a1a7b76f5/google/cloud/pubsub/internal/batching_publisher_connection.cc#L57

If we aren't going to `std::move()` the message in, we should be using a `pubsub::Message const& m`. That saves a copy in the no-op cases.